src/Blazor.FlexGrid/DataSet/TableDataSet.cs

```diff
@@ -132,7 +141,83 @@ public void CancelEditation()
         private void LoadFromQueryableSource()
         {
             PageableOptions.TotalItemsCount = ApplyDeletedConditionToQueryable(source).Count();
-            Items = ApplyFiltersToQueryable(source).ToList();
+            if (!GroupingOptions.IsGroupingActive)
+            {
+                Items = ApplyFiltersToQueryable(source).ToList();
+            }
+            else
+            {
+                GroupedItems = ApplyFiltersWithGroupingToQueryable(source);
+            }
+        }
+
+        private IQueryable<GroupItem<TItem>> ApplyFiltersWithGroupingToQueryable(IQueryable<TItem> source)
+        {
+            try
+            {
+                var groupedItems = this.GroupItems<TItem>(source);
+                groupedItems = ApplyPagingToGroupedQueryable(groupedItems);
+                groupedItems = ApplySortingToGroupedQueryable(groupedItems);
+
+
+                if (this.GroupedItems == null)
+                {
+
+                    return groupedItems.Select(item => new GroupItem<TItem>(item.Key, item.Items) { IsCollapsed = true });
+                }
+                else
+                {
+
+                    //Join query with pre-existing GroupItems collection in order not to lose "IsCollapsed" values
+                    string defaultStrValueIfNull = Guid.NewGuid().ToString();
```

Comment:
When this scenarion is relevant please? Bacause you cannot have same group in separate I am right?

Reply:
When sorting a column while grouping is active, this extra join query allows unfold group of rows to remain unfold.